packages/sidecar/src/utils/docker-compose.ts

```diff
@@ -0,0 +1,199 @@
+  }
+
+  const match = version.match(/Docker version ([\d.]+)/);
+  if (!match) {
+    return null;
+  }
+
+  const versionNumber = match[1];
+  const valid = isVersionGreaterOrEqual(versionNumber, DOCKER_MIN_VERSION);
+
+  return { version: versionNumber, valid };
+}
+
+/**
+ * Detect which compose command to use and its version
+ * Follows the logic from sentry/self-hosted
+ */
+function detectComposeCommand(): { command: string[]; version: string } | null {
+  const composePluginVersion = tryExec("docker compose version --short");
+  const composeStandaloneVersion = tryExec("docker-compose version --short");
+
+  if (!composePluginVersion && !composeStandaloneVersion) {
+    return null;
+  }
+
+  if (!composePluginVersion && composeStandaloneVersion) {
+    return { command: ["docker-compose"], version: composeStandaloneVersion };
+  }
+
+  if (composePluginVersion && !composeStandaloneVersion) {
+    return { command: ["docker", "compose"], version: composePluginVersion };
+  }
+
+  // Use standalone if it's newer, otherwise prefer plugin
+  const pluginVer = composePluginVersion!.replace(/^v/, "");
+  const standaloneVer = composeStandaloneVersion!.replace(/^v/, "");
+
+  if (isVersionGreaterOrEqual(standaloneVer, pluginVer)) {
+    return { command: ["docker-compose"], version: composeStandaloneVersion! };
+  }
+
+  return { command: ["docker", "compose"], version: composePluginVersion! };
+}
+
+/**
+ * Generate the override YAML for injecting Spotlight environment variables
+ */
+function generateSpotlightOverrideYaml(serviceNames: string[], port: number): string {
+  const services: Record<string, unknown> = {};
+
+  // We use host.docker.internal to ensure the container can access the host machine
+  // for backend services that need to access the host machine to send events to the Sidecar.
+  const spotlightUrl = `http://${DOCKER_HOST_INTERNAL}:${port}/stream`;
+  const publicSpotlightUrl = `http://localhost:${port}/stream`;
+
+  for (const serviceName of serviceNames) {
+    services[serviceName] = {
+      environment: [
+        `SENTRY_SPOTLIGHT=${spotlightUrl}`,
+        `NEXT_PUBLIC_SENTRY_SPOTLIGHT=${publicSpotlightUrl}`, // This is needed for Next.js
+        "SENTRY_TRACES_SAMPLE_RATE=1",
+      ],
+      extra_hosts: [`${DOCKER_HOST_INTERNAL}:${DOCKER_HOST_GATEWAY}`],
+    };
+  }
+
+  return stringifyYaml({ services });
+}
+
+/**
+ * Build the Docker Compose command with Spotlight environment injection
+ */
+export function buildDockerComposeCommand(
+  config: DockerComposeConfig,
+  port: number,
+): { cmdArgs: string[]; dockerComposeOverride: string } {
+  const cmdArgs = [...config.command];
+
+  cmdArgs.push("-f", config.composeFile);
+
+  if (config.overrideFile) {
+    cmdArgs.push("-f", config.overrideFile);
+  }
+
+  cmdArgs.push("-f", "-");
+
+  cmdArgs.push("up");
+
+  const dockerComposeOverride = generateSpotlightOverrideYaml(config.serviceNames, port);
+
+  return { cmdArgs, dockerComposeOverride };
```

Comment:
To make this more generic (in case we support other commands similar to this in the future), I'd make the return values `cmdArgs` and `stdin`